plugins/SpectrumAnalyzer/SaProcessor.h

```diff
@@ -72,15 +78,27 @@ class SaProcessor
 	float getAmpRangeMin(bool linear = false) const;
 	float getAmpRangeMax() const;
 
-	// data access lock must be acquired by any friendly class that touches
-	// the results, mainly to prevent unexpected mid-way reallocation
+	// Reallocation lock prevents the processor from changing size of its buffers.
+	// It is used to keep consistent bin-to-frequency mapping while drawing the
+	// spectrum and to make sure reading side does not find itself out of bounds.
+	// The processor is meanwhile free to work on another block.
+	QMutex m_reallocationAccess;
```

Comment:
Why does it make sense to lock the reallocation mutex withouth the data mutex? When you access the buffers, you always access data?

Imagine `SaSpectrumView::drawSpectrum` reads data. It locks the realloc mutex, but the data is meanwhile changed by `SaProcessor`, which locks the data mutex. Can this happen?